### PR TITLE
[now-python] Set PIP_USER environment variable

### LINKED ISSUE #3089: [now dev] fails with @now/python on Debian/Ubuntu
Ubuntu, since Xenial, contains the patched version of `pip` with `--user` and `--ignore-installed` flag active by default. In combination with the `--prefix` option, from the `@now/python` runtime, this always raises an error `distutils.errors.DistutilsOptionError: can't combine user with prefix, exec_prefix/home, or install_(plat)base`.

This can be solved by running `now dev` as a superuser or using the non-system version of `pip`.

For solving this in the runtime either `--system` or `--no-user` (depends on `pip` version) option must be provided. However, `export PIP_USER=0 PIP_IGNORE_INSTALLED=0` seems more applicable.

There are also similar issues: #3023 and #2853

### PR DESCRIPTION
**Pull request by numrut**
Disables `--user` parameter of the `pip` utility, which is forcibly set under the hood on Debian systems and causes an error in the `pipInstall` function:

`distutils.errors.DistutilsOptionError: can't combine user with prefix, exec_prefix/home, or install_(plat)base`

Fixes #3089

### CODE REVIEW

File: packages/now-python/src/index.ts [hunk 1]
```diff
@@ -16,6 +16,14 @@ import {
...
 async function pipInstall(pipPath: string, workDir: string, ...args: string[]) {
   const target = '.';
+  // See: https://github.com/pypa/pip/issues/4222#issuecomment-417646535
+  //
+  // Disable installing to the Python user install directory, which is
+  // the default behavior on Debian systems and causes error:
+  //
+  // distutils.errors.DistutilsOptionError: can't combine user with
+  // prefix, exec_prefix/home, or install_(plat)base
+  process.env['PIP_USER'] = 0;
```
Comment (styfle) [inline packages/now-python/src/index.ts]:
```suggestion
  process.env.PIP_USER = '0';
```

File: packages/now-python/src/index.ts [hunk 2]
```diff
@@ -16,6 +16,14 @@ import {
 
 async function pipInstall(pipPath: string, workDir: string, ...args: string[]) {
   const target = '.';
+  // See: https://github.com/pypa/pip/issues/4222#issuecomment-417646535
```
Comment (styfle) [inline packages/now-python/src/index.ts:19]: This linked issue seems to indicate the correct env var is `PIP_IGNORE_INSTALLED`, not `PIP_USER`

Comment (numrut) [reply packages/now-python/src/index.ts:19]: the patch by default sets two parameters: `--user` and `--ignore-installed`. Their values can be overwritten by `PIP_USER` and `PIP_IGNORE_INSTALLED` environment variables, respectively.

The error is caused by the `--user` flag, so `PIP_USER` var should be used.

**Comment by styfle** [comment]
Do you have a Dockerfile that can reproduce the original error?

Our CI uses the `circleci/node` image but that wasn't affected by this bug so I want to check to make sure this really solves the problem for other distros.

**Review by styfle — APPROVED** [review]
Thanks!

**Pull request merged**